pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -725,6 +731,16 @@ func (pls *PipelineStatus) IsHealthy() bool {
 	}
 }
 
+// MarkServiceNotHealthy marks the Daemon, SideInput and Vertex Service as not healthy with the given reason and message.
```

Comment:
We don't necessarily need to add concrete service names in the comment, in the future we might remove some of them or add more, in that case, we don't want to come back here and update the comment.

```suggestion
// MarkServiceNotHealthy marks a service as not healthy with the given reason and message.
```